metrics/ci_integrations/lib/client/github_actions/github_actions_client.dart

```diff
@@ -0,0 +1,319 @@
+  /// [repositoryName] is `null` or empty.
+  GithubActionsClient({
+    @required this.githubApiUrl,
+    @required this.repositoryOwner,
+    @required this.repositoryName,
+    this.authorization,
+  }) {
+    StringValidator.checkNotNullOrEmpty(githubApiUrl, name: 'githubApiUrl');
+    StringValidator.checkNotNullOrEmpty(repositoryOwner,
+        name: 'repositoryOwner');
+    StringValidator.checkNotNullOrEmpty(repositoryName, name: 'repositoryName');
+  }
+
+  /// Returns the API base path to use in HTTP requests.
+  String get basePath {
+    return '$githubApiUrl/repos/$repositoryOwner/$repositoryName/actions/';
+  }
+
+  /// Creates basic [Map] with headers for HTTP requests.
+  ///
+  /// If provided authorization method is not `null` then adds the result
+  /// of [AuthorizationBase.toMap] method to headers.
+  Map<String, String> get headers {
+    return <String, String>{
+      HttpHeaders.acceptHeader: GithubActionsConstants.acceptHeader,
+      if (authorization != null) ...authorization.toMap(),
+    };
+  }
+
+  /// A method for handling Github-specific HTTP responses.
+  ///
+  /// Awaits [responseFuture] and handles the result. If either the provided
+  /// future throws or [HttpResponse.statusCode] is not equal to [HttpStatus.ok]
+  /// this method results with [InteractionResult.error]. Otherwise, delegates
+  /// processing the response to the given [responseProcessor] callback.
+  Future<InteractionResult<T>> _handleResponse<T>(
+    Future<Response> responseFuture,
+    ResponseProcessingCallback<T> responseProcessor,
+  ) async {
+    try {
+      final response = await responseFuture;
+
+      if (response.statusCode == HttpStatus.ok) {
+        final body = jsonDecode(response.body) as Map<String, dynamic>;
+        final headers = response.headers;
+        return responseProcessor(body, headers);
+      } else {
+        final reason = response.body == null || response.body.isEmpty
+            ? response.reasonPhrase
+            : response.body;
+        return InteractionResult.error(
+          message: 'Failed to perform an operation with code '
+              '${response.statusCode}. Reason: $reason',
+        );
+      }
+    } catch (error) {
+      return InteractionResult.error(
+        message: 'Failed to perform an operation. Error details: $error',
+      );
+    }
+  }
+
+  /// Retrieves a [WorkflowRunsPage] by the given [workflowIdentifier].
+  ///
+  /// The [workflowIdentifier] is either a workflow id or a name of the file
+  /// that defines the workflow.
+  ///
+  /// [status] is used as a filter query parameter to define the
+  /// [WorkflowRun.status] of runs to fetch.
+  ///
+  /// [perPage] is used for limiting the number of runs and pagination in pair
+  /// with the [page] parameter. It defaults to `10` and is limited to `100`.
+  /// The greater values than `100` are ignored and considered as a maximum.
+  ///
+  /// [page] is used for pagination and defines a page of runs to fetch.
```

Comment:
```suggestion
  /// A [status] is used as a filter query parameter to define the
  /// [WorkflowRun.status] of runs to fetch.
  ///
  /// A [perPage] is used for limiting the number of runs and pagination in pair
  /// with the [page] parameter. It defaults to `10` and is limited to `100`.
  /// The greater values than `100` are ignored and considered as a maximum.
  ///
  /// A [page] is used for pagination and defines a page of runs to fetch.
```
Please check other methods as well to be documented in the same way.